### PR TITLE
Deprecate in-place inversion

### PR DESCRIPTION
**Pull request by albi3ro**
This PR deprecates in-place inversion set by either:

```
>>> op.inv()
>>> op.inverse = value
```

We will allow this methods and properties for another two releases till v0.28, when we will delete all in-place inversion infrastructure.

### CODE REVIEW

File: doc/releases/changelog-dev.md
Comment (antalszava) [inline doc/releases/changelog-dev.md:140]: For non-unitaries (e.g., channels) there is a difference between the inverse and adjoint. What could users use going forward there to get the inverse for such operators? Worth mentioning.

If we've found that getting the inverse of non-unitaries is not a use case worth supporting natively in PennyLane, then it is also worth mentioning.

Edit: going fwd I'd guess it's ``op ** -1``.

File: doc/releases/changelog-dev.md
Comment (antalszava) [inline doc/releases/changelog-dev.md:147]: Some sort of code block is required here (e.g., `pycon` block), rendering if off atm:

image

File: pennylane/operation.py
Comment (antalszava) [inline pennylane/operation.py:1489]: Is this required here? We seem to be using the setter in line 1490, doesn't that statement already emit the warning?

**Review by antalszava — CHANGES_REQUESTED** [review]
The logic overall checks out, requesting changes in the changelog (rendering). Also, curious if we need a warning in the `inv` method.

File: pennylane/operation.py
Comment (albi3ro) [reply pennylane/operation.py:1489]: Good point. Removed the one in `inv()`.

File: doc/releases/changelog-dev.md
Comment (albi3ro) [reply doc/releases/changelog-dev.md:148]: added.

File: doc/releases/changelog-dev.md
Comment (antalszava) [inline doc/releases/changelog-dev.md:299]: Or comparable, as per the previous comment.
```suggestion
  use `qml.adjoint` or exponentiating the operator to the `-1`th power instead. Support for these methods will remain till v0.28.
```

**Review by antalszava — CHANGES_REQUESTED** [review]
Thank you! :) One small thing before proceeding.

File: doc/releases/changelog-dev.md
Comment (albi3ro) [reply doc/releases/changelog-dev.md:299]: Actually adding a `qml.pow` constructor in PR #3029 .  That will be easier to reference here than `op**-1`.  Should be a quick PR, then we can update this after that PR is merged.

Comment (antalszava) [reply doc/releases/changelog-dev.md:299]: Can we merge this as suggested and #3029 will update the wording? That way we're on the safe side.

Comment (albi3ro) [reply doc/releases/changelog-dev.md:299]: I can do that.  If this PR is merged first, I'll update the wording of the warning in #3029.

**Review by antalszava — APPROVED** [review]
Looks good :+1: Approving with the note of making sure the changelog reflects all preferred alternatives.

**Pull request merged**
